core-web/libs/edit-content/src/lib/fields/dot-edit-content-wysiwyg-field/components/dot-wysiwyg-tinymce/dot-wysiwyg-tinymce.component.spec.ts

```diff
@@ -0,0 +1,116 @@
+    let dotWysiwygPluginService: DotWysiwygPluginService;
+    let dotWysiwygTinymceService: SpyObject<DotWysiwygTinymceService>;
+
+    const createComponent = createComponentFactory({
+        component: DotWysiwygTinymceComponent,
+        componentViewProviders: [
+            {
+                provide: ControlContainer,
+                useValue: createFormGroupDirectiveMock()
+            },
+            {
+                provide: DotWysiwygPluginService,
+                useValue: {
+                    initializePlugins: jest.fn()
+                }
+            },
+            mockProvider(DotWysiwygTinymceService, {
+                getProps: jest.fn().mockReturnValue(of(mockSystemWideConfig))
+            })
+        ],
+        providers: [FormGroupDirective, provideHttpClient(), provideHttpClientTesting()]
+    });
+
+    beforeEach(() => {
+        spectator = createComponent({
+            props: {
+                field: WYSIWYG_MOCK
+            } as unknown,
+            detectChanges: false
+        });
+
+        dotWysiwygPluginService = spectator.inject(DotWysiwygPluginService, true);
+        dotWysiwygTinymceService = spectator.inject(DotWysiwygTinymceService, true);
+    });
+
+    it('should initialize editor with correct configuration', () => {
+        const expectedConfiguration = {
+            ...DEFAULT_TINYMCE_CONFIG,
+            ...mockSystemWideConfig,
+            setup: (editor) => dotWysiwygPluginService.initializePlugins(editor)
+        };
+
+        spectator.detectChanges();
+        expect(JSON.stringify(spectator.component.$editorOptions())).toEqual(
+            JSON.stringify(expectedConfiguration)
+        );
+    });
+
+    it('should parse custom props from field variables', () => {
+        const fieldVariables = [
+            {
+                clazz: 'com.dotcms.contenttype.model.field.ImmutableFieldVariable',
+                fieldId: '1',
+                id: '1',
+                key: 'tinymceprops',
+                value: '{ "toolbar1": "undo redo"}'
+            }
+        ];
+
+        spectator = createComponent({
+            props: {
+                field: {
+                    ...WYSIWYG_MOCK,
+                    fieldVariables
+                }
+            } as unknown,
+            detectChanges: false
+        });
+
+        spectator.detectChanges();
+
+        expect(JSON.stringify(spectator.component.$editorOptions())).toEqual(
+            JSON.stringify({
+                ...DEFAULT_TINYMCE_CONFIG,
+                ...mockSystemWideConfig,
+                ...{ toolbar1: 'undo redo' },
+                setup: (editor) => dotWysiwygPluginService.initializePlugins(editor)
+            })
+        );
+    });
+
+    it('should set the system wide props', () => {
+        const newSystemWideConfig = { systemWideOption: 'new_value' };
+
+        jest.spyOn(dotWysiwygTinymceService, 'getProps').mockReturnValue(of(newSystemWideConfig));
```

Comment:
Here is a way to avoid manual spying using the mockProvider utility from @ngneat/spectator

Reply:
Actually, I am using mockProvider. 
In this step, I changed these values to a new` 'new_value' `to double-check if the options were generated correctly.

Reply:
If you are using the `mockProvider` then you can avoid creating the manual spy and replace with this:

`dotWysiwygTinymceService. getProps().mockReturnValue(of(newSystemWideConfig));`